elephant/joint_isi_dithering_class.py

```diff
@@ -0,0 +1,462 @@
+'''
+Module to generate surrogates of spike trains by using the joint-ISI dithering.
+Features are provided to separate the preprocessing from the main process.
+
+Original implementation by: Peter Bouss [p.bouss@fz-juelich.de]
+:copyright: Copyright 2019 by the Elephant team, see `doc/authors.rst`.
+:license: Modified BSD, see LICENSE.txt for details.
+'''
+
+
+import numpy as np
+import quantities as pq
+import neo
+
+import elephant.spike_train_surrogates as surr
+import elephant.statistics as stats
+from scipy.ndimage import gaussian_filter
+
+try:
+    import matplotlib.pyplot as plt
+    HAVE_PLT = True
+except ImportError:
+    HAVE_PLT = False
+
+class Joint_ISI_Space:
+    '''
+    The class :class:`Joint_ISI_Space` is implemented for Joint-ISI dithering
+    as a continuation of the ideas of Louis et al. (2010) and Gerstein (2004).
+
+    When creating an class instance all necessary preprocessing steps are done,
```

Comment:
This is principle should be changed. Spiketrains with too low firing rate should be dithered by the joint isi anyway: maybe adapting the width of the gaussian kernel accordingly (?)

Reply:
Here I'm still thinking of how to do it. And we can maybe discuss within the next days, how to procede with this. I have some ideas, regarding this.